app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/triage/multipledisposals/asset_type_for_non_uk_residents.scala.html

```diff
@@ -72,56 +75,78 @@
   }
 }
 
-@options = @{
-    List(
-      RadioOption(
-        label = messages(s"$key.Residential"),
-        content = None,
-        optionHelpText = Some(Html(messages(s"$key.Residential.helpText")))
-      ),
-      RadioOption(
-        label = messages(s"$key.NonResidential"),
-        content = None,
-        optionHelpText = Some(Html(messages(s"$key.NonResidential.helpText")))
-      ),
-      RadioOption(
-        label = messages(s"$key.MixedUse"),
-        content = None,
-        optionHelpText = Some(Html(messages(s"$key.MixedUse$userKey.helpText")))
-      ),
-      RadioOption(
-        label = messages(s"$key$userKey.IndirectDisposal"),
-        content = None,
-        optionHelpText = Some(Html(messages(s"$key.IndirectDisposal.helpText")))
-      )
-    )
-}
-
-@mainTemplate(title = title, userType = request.userType, hasErrors = hasErrors) {
-
-    @backLinkComponent(backLink)
+@selected = @{form.value.map(_.map(toIndex)).getOrElse(List.empty)}
```

Comment:
we have a better way to evaluate the checked state of checkboxes or radios that does not involve @selected or @toIndex - in this case we have a list of asset types as the form value, we can pattern match the form.value and if it's a list then check if the list contains the asset type e.g. `x.contains(AssetType.Residential)`